eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala

```diff
@@ -247,11 +247,21 @@ class Setup(datadir: File,
           feeratesRetrieved.trySuccess(Done)
       })
       _ <- feeratesRetrieved.future
+      (zmqBlocks_opt, zmqTxs_opt) <- bitcoin match {
+        case Bitcoind(client) => client.invoke("getzmqnotifications").map {
+          case JArray(zmqInfos) =>
+            val blocksAddress = zmqInfos.find(_ \ "type" == JString("pubrawblock")).map(_.\("address").extract[String])
+            val txsAddress = zmqInfos.find(_ \ "type" == JString("pubrawtx")).map(_.\("address").extract[String])
+            (blocksAddress, txsAddress)
+          case _ => (None, None)
+        }
+        case _ => Future.successful((None, None))
+      }
 
       watcher = bitcoin match {
         case Bitcoind(bitcoinClient) =>
-          system.actorOf(SimpleSupervisor.props(Props(new ZMQActor(config.getString("bitcoind.zmqblock"), Some(zmqBlockConnected))), "zmqblock", SupervisorStrategy.Restart))
-          system.actorOf(SimpleSupervisor.props(Props(new ZMQActor(config.getString("bitcoind.zmqtx"), Some(zmqTxConnected))), "zmqtx", SupervisorStrategy.Restart))
+          system.actorOf(SimpleSupervisor.props(Props(new ZMQActor(zmqBlocks_opt.get, Some(zmqBlockConnected))), "zmqblock", SupervisorStrategy.Restart))
```

Comment:
This looks like it will throw cryptic exceptions when bitcoin core is started with zmq disabled.

Reply:
At 3d37e08 we check beforehand to avoid the cryptic exception in case the addresses are missing

Reply:
using `.get` on an `Option` is almost always a code smell (I don't think there is another instance of this in our code ?)

Reply:
I'll have a look on how to improve that. Yes it is a bit of a code smell but I think it's acceptable only if there is a `nonEmpty` check very close by (because of readability), unfortunately we do have several instances of `.get` in the codebase.